### DISCUSSION TITLE
Deploy on Release making preview deployments.

### DISCUSSION THREAD

**Discussion by debater-coder** [vercel/community#1268] [Help]
### Summary

I have been following [this guide](https://vercel.com/guides/can-you-deploy-based-on-tags-releases-on-vercel) on deploying based on releases, but whenever I do a post request to my deploy hook, it triggers a preview deployment, and does not make a production deployment. How do I make it so that whenever I create a release, the contents of the main branch are deployed to production?

**Comment by amyegan** [comment]
Do you have `main` configured as both the [Deploy Hook](https://vercel.com/docs/concepts/git/deploy-hooks#creating-a-deploy-hook) branch and the [Production Branch](https://vercel.com/docs/concepts/git#production-branch) in Project Settings?

**Reply by debater-coder** [reply]
I did not since from what I understood the guide says not to. I've changed that now, and will check if it works. Wouldn't that mean that every commit on `main` is deployed to production (I only want to deploy when the deploy hook is triggered not on every commit)?

**Reply by amyegan** [reply]
To avoid automatic deployment, you can set [`git.deploymentEnabled`](https://vercel.com/docs/project-configuration#git-configuration/git-deployment-enabled) to `false` in a project config `vercel.json`

**Reply by arbourd** [reply]
I'm doing something similar but I want to disable all automatic deploys to `main` with [`git.deploymentEnabled`](https://vercel.com/docs/project-configuration#git-configuration/git-deployment-enabled). 

Will preview deploys for pull-requests still work against `main`?

**Reply by amyegan** [reply]
If you set `deploymentEnabled` to false for the `main` branch, then _all_ auto-deployments against that branch will be disabled (but deploy hooks and CLI deployments will still be allowed). 

Preview deployments would also be skipped, so you would need to set up a preview deployment action to get preview deployments. In that case, I would recommend switching to the CLI as in the [use GitHub Actions with Vercel](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel) example. That will give you more control over your deployments.

**Reply by arbourd** [reply]
Is there any way to set the `deploymentEnabled` functionality on the project itself, rather than the `vercel.json`? The reason I ask is we have two projects from the same source repository with slightly different implementations, and using the `vercel.json` would affect both projects.

**Reply by amyegan** [reply]
🤔 That's tricky. You could use Ignored Build Step to skip `main` on just that project, but you would also need to switch to using the CLI instead of the deploy hook in your GitHub Actions script.

**Reply by lukevella** [reply]
I seem to be having the exact same problem. As @arbourd said the documentation is quite specific in stating:

> Create an unused git branch, go to your Vercel settings, and change your production branch to the unused branch. This is to ensure Vercel isn’t running a new deployment on every commit to your main branch, as we only want to deploy releases or tags to production.

[Source](https://vercel.com/guides/can-you-deploy-based-on-tags-releases-on-vercel)

The expected behavior seems to be that the deploy hook should create a production build, otherwise it is useless since a preview built is already automatically created when we push to a branch.
